File: trunk/emsi/asp-net-app/WebEMSOF--feature-link-spreadsheet-rows-to-detail/component/db/Class_db_fiscal_years.pas
unit Class_db_fiscal_years;

interface

uses
  borland.data.provider,
  Class_db;

type
  TClass_db_fiscal_years = class(TClass_db)
  private
    { Private Declarations }
  public
    constructor Create;
    function IdOfCurrent: string;
    function IdOfDesignator(designator: string): string;
  end;

implementation

constructor TClass_db_fiscal_years.Create;
begin
  inherited Create;
  // TODO: Add any constructor code here
end;

function TClass_db_fiscal_years.IdOfCurrent: string;
begin
  self.Open;
  IdOfCurrent := borland.data.provider.bdpcommand.Create('select max(id) from fiscal_year',connection).ExecuteScalar.tostring;
  self.Close;
end;

function TClass_db_fiscal_years.IdOfDesignator(designator: string): string;
begin
  self.Open;
  IdOfDesignator := borland.data.provider.bdpcommand.Create
    ('select id from fiscal_year where designator = "' + designator + '"',connection).ExecuteScalar.tostring;
  self.Close;
end;

end.

File: trunk/emsi/asp-net-app/WebEMSOF/component/db/Class_db_fiscal_years.cs
using MySql.Data.MySqlClient;
using System;

using System.Web.UI.WebControls;
using Class_db;
namespace Class_db_fiscal_years
{
    public class TClass_db_fiscal_years: TClass_db
    {
        //Constructor  Create()
        public TClass_db_fiscal_years() : base()
        {
            // TODO: Add any constructor code here

        }
        public void BindListControl(object target)
        {
            this.Open();
            ((target) as ListControl).DataSource = new MySqlCommand("select id,designator from fiscal_year order by id desc", this.connection).ExecuteReader();
            ((target) as ListControl).DataValueField = "id";
            ((target) as ListControl).DataTextField = "designator";
            ((target) as ListControl).DataBind();
            this.Close();
        }

        public string DesignatorOfCurrent()
        {
            string result;
            this.Open();
            result = new MySqlCommand("select designator from fiscal_year order by id desc limit 1", this.connection).ExecuteScalar().ToString();
            this.Close();
            return result;
        }

        public string IdOfCurrent()
        {
            string result;
            this.Open();
            result = new MySqlCommand("select max(id) from fiscal_year", this.connection).ExecuteScalar().ToString();
            this.Close();
            return result;
        }

        public string IdOfDesignator(string designator)
        {
            string result;
            this.Open();
            result = new MySqlCommand("select id from fiscal_year where designator = \"" + designator + "\"", this.connection).ExecuteScalar().ToString();
            this.Close();
            return result;
        }

        public void StartNew(string designator)
        {
            this.Open();
            new MySqlCommand("insert ignore fiscal_year set designator = \"" + designator + "\"", this.connection).ExecuteNonQuery();
            this.Close();
        }

    } // end TClass_db_fiscal_years

}
